charts/cortx/templates/_helpers.tpl

```diff
@@ -180,6 +180,46 @@ Return the Motr confd endpoint port
 21001
 {{- end -}}
 
+{{/*
+Return the number of StatefulSets required to fullfil the containerGroupSize to CVG mapping defined by the user
+This is calculated by (Number of CVGs in solution.yaml) / ({}.containerGroupSize).
+If CVGs are defined, the minimum value this should return is 1.
+If CVGs are not defined, this should return 0.
+*/}}
+{{- define "cortx.data.statefulSetCount" -}}
+{{- if eq (len .Values.cortxdata.cvgs) 0 -}}
+0
+{{- else -}}
+{{- printf "%d" (max 1 (ceil (div (len .Values.cortxdata.cvgs) (.Values.cortxdata.motr.containerGroupSize|int)))) -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns the maximally-allowed value for ContainerGroupSize,
+which prevents the containerGroupSize parameter from being larger than the number of CVGs.
+*/}}
+{{- define "cortx.data.validatedContainerGroupSize" -}}
+{{- printf "%d" (min (len .Values.cortxdata.cvgs) (.Values.cortxdata.motr.containerGroupSize|int)) -}}
+{{- end -}}
+
+{{/*
+Returns the string used to group CORTX Data StatefulSets.
+NOTE: Until CORTX-32368 is resolved, this needs to be a single character due to FQDN length issues.
+*/}}
+{{- define "cortx.data.groupPrefix" -}}
+g
+{{- end -}}
+
+{{/*
+Returns the fullname of the CORTX Data StatefulSet with group suffix.
+Must be called with input scope of a Dictionary with the following keys and values:
+- .root = $
+- .sts_index = Iterator index of all CORTX Data StatefulSets
+*/}}
+{{- define "cortx.data.groupFullname" -}}
+{{- printf "%s-%s%02d" (include "cortx.data.fullname" .root) (include "cortx.data.groupPrefix" $) .sts_index -}}
```

Comment:
Based on comments I've seen, wasn't there a desire to be able to run 106 cvg groups? If that's true, what happens with the `%02` if it's 3 digits, does one get truncated?

Reply:
Once the long FQDN issues get resolved, I can update this to be a three character ("000") string. I don't foresee too many deployments reaching over 100 StatefulSets deployed, but having the explicit naming support for it is better than not. For now, with the limitations that we have, this formatting provides a robust enough deployment pattern for the time being.

Reply:
Digging in further, it looks like it's not needed per kubectl already sorted as desired (instead of standard numeric sorting), so removing the padding for now.